native/packages/react-native-bpk-component-text/src/BpkText.js

```diff
@@ -23,97 +23,147 @@ import {
   colorGray700,
   fontFamily,
   fontFamilyEmphasize,
+  fontFamilyHeavy,
   textEmphasizedFontWeight,
+  textHeavyFontWeight,
+  letterSpacingCaps,
+  textCapsFontSize,
+  textCapsFontWeight,
+  letterSpacingXs,
   textXsFontSize,
   textXsFontWeight,
+  letterSpacingSm,
   textSmFontSize,
   textSmFontWeight,
+  letterSpacingBase,
   textBaseFontSize,
   textBaseFontWeight,
+  letterSpacingLg,
   textLgFontSize,
   textLgFontWeight,
+  letterSpacingXl,
   textXlFontSize,
   textXlFontWeight,
+  letterSpacingXxl,
   textXxlFontSize,
   textXxlFontWeight,
+  letterSpacingXxxl,
+  textXxxlFontSize,
+  textXxxlFontWeight,
 } from 'bpk-tokens/tokens/base.react.native';
 import { Text, Platform, StyleSheet } from 'react-native';
 
 import { shouldApplyFontWeightFix } from './font-weight-fix';
-import { propTypes, defaultProps, type Props } from './common-types';
+import {
+  propTypes,
+  defaultProps,
+  type Props,
+  WEIGHT_STYLES,
+} from './common-types';
 
 const TEXT_TOKENS = {
+  letterSpacingCaps,
+  textCapsFontSize,
+  textCapsFontWeight,
+  letterSpacingXs,
   textXsFontSize,
   textXsFontWeight,
+  letterSpacingSm,
   textSmFontSize,
   textSmFontWeight,
+  letterSpacingBase,
   textBaseFontSize,
   textBaseFontWeight,
+  letterSpacingLg,
   textLgFontSize,
   textLgFontWeight,
+  letterSpacingXl,
   textXlFontSize,
   textXlFontWeight,
+  letterSpacingXxl,
   textXxlFontSize,
   textXxlFontWeight,
+  letterSpacingXxxl,
+  textXxxlFontSize,
+  textXxxlFontWeight,
 };
 
 const getStyleByTextStyle = textStyle => {
   const camelCasedStyle = textStyle[0].toUpperCase() + textStyle.slice(1);
 
   const {
+    [`letterSpacing${camelCasedStyle}`]: letterSpacing,
     [`text${camelCasedStyle}FontSize`]: fontSize,
     [`text${camelCasedStyle}FontWeight`]: fontWeight,
   } = TEXT_TOKENS;
 
   return {
     color: colorGray700,
+    letterSpacing,
     fontFamily,
     fontSize,
     fontWeight,
   };
 };
 
-const getEmphasizeProperties = () => {
+const getEmphasizeProperties = weight => {
```

Comment:
This needs to be smarter because combinations like `<BpkText textStyle="lg" weight="heavy">` are not valid and should probably cause a prop type warning and render as regular

Reply:
I guess this logic should be handled by custom prop types rather than this function?
I think this function should be able to assume that, by the time it's being called, the values are valid.
Let me know if you disagree

Reply:
Done